test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatAddPkgTests.cs

```diff
@@ -745,6 +746,123 @@ public async Task AddPkg_ConditionalAddWithUserInputFramework_Success(string pac
+            {
+                var projectFrameworks = "net472";
+                var packageFrameworks = "net472; netcoreapp2.0";
+                var projectA = XPlatTestUtils.CreateProject(ProjectName, pathContext, projectFrameworks);
+                var packageX = XPlatTestUtils.CreatePackage(frameworkString: packageFrameworks);
+                var packageXPath = Path.Combine(pathContext.WorkingDirectory, "packageX");
+                var mySourcePath = Path.Combine(pathContext.WorkingDirectory, "MySource");
+
+                Directory.CreateDirectory(mySourcePath);
+
+                // Generate Package
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    packageXPath,
+                    PackageSaveMode.Defaultv3,
+                    packageX);
+
+                string originalPackageXNupkg = Path.Combine(packageXPath, packageX.Id, packageX.Version, $"{packageX.Id}.{ packageX.Version}.nupkg");
+                string mySourcePackageXNupkg = Path.Combine(mySourcePath, $"{packageX.Id}.{ packageX.Version}.nupkg");
+
+                if (!File.Exists(originalPackageXNupkg))
+                {
+                    // On linux it's created in different place.
+                    string[] nupkgFiles = Directory.GetFiles(packageXPath, $"*.nupkg", SearchOption.AllDirectories);
+                    Assert.True(nupkgFiles.Count() == 1);
+                    originalPackageXNupkg = nupkgFiles[0];
+                }
+
+                // We're testing case we have only Nupkg file created at top.
+                File.Move(originalPackageXNupkg, mySourcePackageXNupkg);
+
+                // Remove packageXPath directory to prove we're not using nuget package structured folders.
+                var failedDeletes = new List<string>();
+                LocalResourceUtils.DeleteDirectoryTree(packageXPath, failedDeletes);
+
+                // Since user is not inputing a version, it is converted to a " * " in the command
+                var packageArgs = XPlatTestUtils.GetPackageReferenceArgs(packageX.Id, "*",
+                    projectA,
+                    sources: mySourcePath);
+
+                var commandRunner = new AddPackageReferenceCommandRunner();
+
+                // Act
+                var result = commandRunner.ExecuteCommand(packageArgs, MsBuild)
+                    .Result;
+
+                // Assert
+                Assert.Equal(0, result);
+                Assert.True(failedDeletes.Count() == 0);
+
+                var ridlessTarget = projectA.AssetsFile.Targets.Where(e => string.IsNullOrEmpty(e.RuntimeIdentifier)).Single();
+                ridlessTarget.Libraries.Should().Contain(e => e.Type == "package" && e.Name == packageX.Id);
+                ridlessTarget.Libraries.Should().Contain(e => e.Version.Equals(NuGetVersion.Parse(packageX.Version)));
+            }
+        }
+
+        [Fact]
+        public async Task AddPkg_WithAdditionalSourceFeed_VersionSpecified_Success()
```

Comment:
Same questions apply here as the above, it's a bit difficult to follow.

Reply:
I'll wait on your review on other unit test.

Reply:
Fixed it and added 2 more unit test for unhappy paths.